Improve build process and github workflows. Verify the docs links.

amstewart opened the pull request:
# Summary

1. Add a top-level project `Makefile` with basic targets like: all, clean, and lint.
2. The `lint` target will run `sphinx linkcheck` on the docs sources.
3. This patchset fixes a couple broken links.
4. Rewrite the .github actions and workflows so that PRs are automatically built  and linted.
5. On each new push into `main`, the docs will be built and automatically pushed to the `gh-pages` ref.
6. Add a `requirements.txt` file to coordinate python dependencies between build environments.
7. Remove the unused `make.bat` file.


# Testing

* [Example PR Check job run](https://github.com/amstewart/nilrt-docs/actions/runs/17505138114/job/49727002725?pr=1)
 * You can see an example of the deployment action [on my repo](https://github.com/amstewart/nilrt-docs/actions/runs/17505210747/job/49727243351).
    * And the [published pages here](https://amstewart.github.io/nilrt-docs/)


# Process

This patchset is based on #87.

Commit:
docs: remove make.bat

We do not support building the sources on Windows, so remove the
unneeded batch file.

Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

Commit:
Add a top-level Makefile

Add a top-level Makefile for convenience.

Update the README with new build instructions.

Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

Commit:
Add a python requirements.txt file

Sphinx documentation output can vary substantially between sphinx build
environments, if there is no way to coordinate python module versions
between build systems. Use a requirements.txt file to do that
coordination.

Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

Commit:
Use sphinx linkcheck to check link validity

Add a Makefile target to run the Sphinx linkcheck 'builder', which
actually checks the validity of the external links in the documentation.

Fixup a couple broken links found in the documentation.

Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

Commit:
.github: add a pr-check workflow

The PR check should try to build the project and lint it. If either
fails, the PR shouldn't be accepted.

Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

Commit:
.github:publish-page.yml: use peaceiris' gh-pages action

Use peaceiris' gh-pages action to create and publish the gh-pages,
because it can do so in an automated fashion (does not require final
approval from an admin.)

When using this action, it automatically adds a .nojekyll file, so
remove the unneeded source file.

Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

chaitu236 reviewed (commented):
<img width="1836" height="458" alt="image" src="https://github.com/user-attachments/assets/a26e81ff-a20a-43c5-9298-d3954e711626" />

Can you rebase your commits and drop the merge commit

Commit:
config_dev_system: fixup too-short header

Resolves docutils compile warning.

Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

Commit:
config_dev_system: fixup gcc versions table

Previous table file path pointed at an obsolete file. Corrected to new
file path.

Signed-off-by: Alex Stewart <alex.stewart@emerson.com>

Commit:
clamav: fixup clamav wiki FAQ link

Signed-off-by: Alex Stewart <alex.stewart@emerson.com>
